core/src/test/java/org/elasticsearch/indices/recovery/RecoveryTests.java

```diff
@@ -120,26 +143,26 @@ public void testRecoveryWithOutOfOrderDelete() throws Exception {
                     IndexMetaData.Builder builder = IndexMetaData.builder(orgReplica.indexSettings().getIndexMetaData());
                     builder.settings(Settings.builder().put(orgReplica.indexSettings().getSettings())
                         .put(IndexSettings.INDEX_TRANSLOG_RETENTION_AGE_SETTING.getKey(), "-1")
-                        .put(IndexSettings.INDEX_TRANSLOG_RETENTION_SIZE_SETTING.getKey(), "-1")
-                    );
+                        .put(IndexSettings.INDEX_TRANSLOG_RETENTION_SIZE_SETTING.getKey(), "-1"));
                     orgReplica.indexSettings().updateIndexMetaData(builder.build());
                     orgReplica.onSettingsChanged();
-                    translogOps = 3; // 2 ops + seqno gaps
+                    translogOps = 5; // 4 ops + seqno gaps (delete #1 is removed)
                 } else {
                     logger.info("--> flushing shard (translog will be retained)");
-                    translogOps = 4; // 3 ops + seqno gaps
+                    translogOps = 6; // 5 ops + seqno gaps
                 }
+                orgReplica.updateGlobalCheckpointOnReplica(1L, "test");
                 flushShard(orgReplica);
```

Comment:
I'm not sure about this - it also creates a new commit point and that is the one we'll use for recovery (because we don't do rollback yet) and it will point at the last translog gen, I think. Are you sure it's ok?